internal/icons/icons.go

```diff
@@ -1,134 +1,24 @@
-// Package icons embeds several PNG icons.
+// Package icons generates and loads icon Gresource.
 package icons
 
 import (
-	"context"
-	"embed"
-	"io/fs"
+	_ "embed"
 	"log"
-	"net/url"
-	"path"
-	"path/filepath"
-	"strings"
-	"sync"
 
-	"github.com/diamondburned/gotk4/pkg/gdk/v4"
-	"github.com/diamondburned/gotk4/pkg/gdkpixbuf/v2"
 	"github.com/diamondburned/gotk4/pkg/gio/v2"
 	"github.com/diamondburned/gotk4/pkg/glib/v2"
-	"github.com/diamondburned/gotkit/gtkutil/imgutil"
-	"github.com/diamondburned/gotkit/gtkutil/textutil"
 )
 
-//go:generate -command pack-icons go run ./pack-icons -s rounded -px 20
-//go:generate pack-icons svg/*.svg alternate_email add_reaction campaign forum chat mark_chat_unread mark_chat_read send
+//go:generate glib-compile-resources gtkcord4.gresource.xml
 
-//go:generate ./svg2png.sh
+//go:embed gtkcord4.gresource
+var Resources []byte
 
-//go:embed png/*.png
```

Comment:
Can this be in `init()`?

Reply:
Sorry, this was supposed to refer to `func LoadResources() {`.

Reply:
As `icons` module is now imported only once, I think it's better to put resource init code in a function and call it manually in main's `init`

Reply:
The codebase already makes liberal use of `init`, so I don't know if there's still a point in refraining to use it, especially when there's no reason to delay the call until `main.init`.